ingestion/src/metadata/ingestion/source/pipeline/databrickspipeline/metadata.py

```diff
@@ -0,0 +1,222 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Databricks pipeline source to extract metadata
+"""
+
+import traceback
+from typing import Any, Iterable, List, Optional
+
+from metadata.generated.schema.api.data.createPipeline import CreatePipelineRequest
+from metadata.generated.schema.api.lineage.addLineage import AddLineageRequest
+from metadata.generated.schema.entity.data.pipeline import (
+    PipelineStatus,
+    StatusType,
+    Task,
+    TaskStatus,
+)
+from metadata.generated.schema.entity.services.connections.metadata.openMetadataConnection import (
+    OpenMetadataConnection,
+)
+from metadata.generated.schema.entity.services.connections.pipeline.databricksPipelineConnection import (
+    DatabricksPipelineConnection,
+)
+from metadata.generated.schema.metadataIngestion.workflow import (
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.models.pipeline_status import OMetaPipelineStatus
+from metadata.ingestion.source.database.databricks.client import DatabricksClient
+from metadata.ingestion.source.pipeline.pipeline_service import PipelineServiceSource
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+STATUS_MAP = {
+    "SUCCESS": StatusType.Successful,
+    "FAILED": StatusType.Failed,
+    "TIMEOUT": StatusType.Failed,
+    "CANCELED": StatusType.Failed,
+    "PENDING": StatusType.Pending,
+    "RUNNING": StatusType.Pending,
+    "TERMINATING": StatusType.Pending,
+    "SKIPPED": StatusType.Failed,
+    "INTERNAL_ERROR": StatusType.Failed,
+}
+
+TASK_MAP = {
+    "notebook_task": "Notebook_Task",
```

Comment:
Should we bring the values into the JSON Schema for Databricks pipelines to have a closed list of supported tasks?

Reply:
So I just went ahead and removed the TASK_MAP. I only really included it to make the formatting look nicer in the UI.

Instead, I updated this process just to find the key named like "*_task" in the task details for a pipeline. You can only have one task type for each task, and I am assuming that all future task types are going to have the same naming convention.

Reply:
sounds good, thanks for explaining